extension/src/implementation.ts

```diff
@@ -0,0 +1,52 @@
+import { Installation } from "@exasol/extension-manager-interface";
+import { ExaAllScriptsRow } from "@exasol/extension-manager-interface/dist/exasolSchema";
+import { ADAPTER_SCRIPT_NAME, Context, IMPORT_SCRIPT_NAME } from "./common";
+
+function findScriptByName(scripts: ExaAllScriptsRow[], name: string): ExaAllScriptsRow | undefined {
+    return scripts.find(script => script.name === name);
+}
+
+export function findInstallations(scripts: ExaAllScriptsRow[], context: Context): Installation[] {
+    const importScript = findScriptByName(scripts, IMPORT_SCRIPT_NAME);
+    const adapterScript = findScriptByName(scripts, ADAPTER_SCRIPT_NAME);
+    if (isValidImportScript(importScript) && isValidAdapterScript(adapterScript)) {
+        return [{
+            name: `${adapterScript.schema}.${adapterScript.name}`, version: context.version, instanceParameters: []
+        }];
+    } else {
+        return [];
+    }
+}
+
+function isValidAdapterScript(script: ExaAllScriptsRow): script is ExaAllScriptsRow {
+    if (!script) {
+        console.log(`Adapter script ${ADAPTER_SCRIPT_NAME} not found`)
```

Comment:
Should this really be written to console, here?
Or is console visible to the human user?

Reply:
This appears only in the stdout of the server process, not the browser :)